supportedBackends/bcs/src/main/scala/cromwell/backend/impl/bcs/BcsJob.scala

```diff
@@ -0,0 +1,196 @@
+package cromwell.backend.impl.bcs
+
+import com.aliyuncs.batchcompute.main.v20151111.BatchComputeClient
+import com.aliyuncs.batchcompute.model.v20151111._
+import com.aliyuncs.batchcompute.pojo.v20151111._
+import cromwell.core.ExecutionEvent
+import cromwell.core.path.Path
+
+import collection.JavaConverters._
+import scala.util.Try
+
+object BcsJob{
+  val BcsDockerImageEnvKey = "BATCH_COMPUTE_DOCKER_IMAGE"
+  val BcsDockerPathEnvKey = "BATCH_COMPUTE_DOCKER_REGISTRY_OSS_PATH"
+}
+
+case class BcsJob(name: String,
+                  description: String,
+                  commandString: String,
+                  packagePath: Path,
+                  mounts: Seq[BcsMount],
+                  envs: Map[String, String],
+                  runtime: BcsRuntimeAttributes,
+                  batchCompute: BatchComputeClient) {
+
+  lazy val lazyDisks = new Disks
+  lazy val lazyConfigs = new Configs
+  lazy val lazyVpc = new VPC
+  lazy val lazyTask = new TaskDescription
+  lazy val lazyJob = new JobDescription
+  lazy val lazyCmd = new Command
+
+  def submit(): Try[String] = Try{
+    val request: CreateJobRequest = new CreateJobRequest
+    request.setJobDescription(jobDesc)
+    val response: CreateJobResponse = batchCompute.createJob(request)
+    val jobId = response.getJobId
+    jobId
+  }
+
+  def getStatus(jobId: String): Try[RunStatus] = {
+    val request: GetJobRequest = new GetJobRequest
+    request.setJobId(jobId)
+    val response: GetJobResponse = batchCompute.getJob(request)
+    val job = response.getJob
+    val status = job.getState
+    val message = job.getMessage
+    val eventList = Seq[ExecutionEvent]()
+    RunStatusFactory.getStatus(jobId, status, Some(message), Some(eventList))
+  }
+
+  def cancel(jobId: String): Unit = {
+    // XXX: Do nothing currently.
+  }
+
+  private[bcs] def systemDisk: Option[SystemDisk] = runtime.systemDisk map { disk =>
+      val systemDisk = new SystemDisk()
+      systemDisk.setType(disk.diskType)
+      systemDisk.setSize(disk.sizeInGB)
+      systemDisk
+  }
+
+  private[bcs] def dataDisk: Option[DataDisk] = runtime.dataDisk map { disk =>
+    val dataDisk = new DataDisk
+    dataDisk.setType(disk.diskType)
+    dataDisk.setSize(disk.sizeInGB)
+    dataDisk.setMountPoint(disk.mountPoint)
+    dataDisk
+  }
+
+  // XXX: maybe more elegant way to reduce two options?
```

Comment:
perhaps
```scala
systemDisk.foreach(lazyDisks.setSystemDisk)
dataDisk.foreach(lazyDisks.setDataDisk)
```

Reply:
Thx, it works.

Reply:
@crisish Can you make sure to fold in the suggestion from @danbills ?

Reply:
@geoffjentry It works but not perfect because I expect to return a None value if both `systemDisk` and `dataDisk` is empty.

Reply:
Ah ok. I was just going by the previous comment